### ISSUE TITLE
unstable connection: disconnects, slow responses, lots of retransmissions

### ISSUE THREAD

**Issue by bertmelis** [me-no-dev/AsyncTCP#27]
I'm struggling to get the lib working for a simple modbus master. I can't even connect to the server.
Wireshark shows a remarkable log:
![image](https://user-images.githubusercontent.com/23474336/48702705-dfd93580-ebf1-11e8-8e7e-d49b4e333d5d.png)

Even though the ESP is already connected, it resends SYN. Also, it takes about 3 seconds before the ESP knows it has connected (wireshark show it about 3 secs sooner).

From time to time, I experience stranger errors like `[E][AsyncTCP.cpp:571] _recv(): 0x3ffbb578 != 0x3ffbb120`

Behaviour is the same for the main branch as the idf-update branch. The _recv() only appears at the idf-update branch.

Edit: when the `_recv()` error appears, there's no callback called, so my app also doesn't know it can move on. As a workaround I need to implement a timer myself.

**Comment by bertmelis** [comment]
So what I believe is happening:
- I call `client.connect()`
- I call `client.connect()` again, assuming it would just return because it is already connecting. However, as my Wireshark analysis shows, it is creating a new connection.
- AsyncClient connects, but it calls my `onConnected` several times: for each call I made to `connect()`.
- After being connected, I start with regular communication.
- Because I only use the "lastest" connection, the other ones get idle and get disconnected by the server.
- AsyncClient treats the disconnect from the frist connection as a disconnect from the active connection.

I though solving this by
```C++
if (!client.connected() && !client.connecting()) {
  client.connect();
}
```

But AsyncClient takes a few seconds before being aware that it is connecting. Setting the priority of the asynctask higher improves, but is still not aware of it's state directly after calling `connect()`. Surely it is a issue with FreeRTOS's task switching latency.

So basically:
Is this intended behaviour and should I deal with this myself?
Or is it an issue that has to be solved in AsyncClient?

**Comment by bertmelis** [comment]
Closing for now...
Probably has something to do with 
- bad WiFi reception
- data coming after disconnect
- ack timeouts
